fastlane/lanes/build.rb

```diff
@@ -80,31 +82,61 @@
     )
   end
 
-  # Builds the app and uploads it to TestFlight / App Store Connect
+  # Builds the WordPress app and uploads it to TestFlight, for beta-testing or final release
   #
   # @option [Boolean] skip_confirm (default: false) If true, avoids any interactive prompt
-  # @option [Boolean] create_gh_release If true, creates a GitHub Release draft after the upload, with zipped xcarchive as artefact
+  # @option [Boolean] skip_prechecks (default: false) If true, don't run the ios_build_prechecks and ios_build_preflight
+  # @option [Boolean] create_release If true, creates a GitHub Release draft after the upload, with zipped xcarchive as artefact
   # @option [Boolean] beta_release If true, the GitHub release will be marked as being a pre-release
   #
   # @called_by CI
-  # @calls build_and_upload_itc
   #
   desc 'Builds and uploads for distribution to App Store Connect'
   lane :build_and_upload_app_store_connect do |options|
-    ios_build_prechecks(
-      skip_confirm: options[:skip_confirm],
-      internal: options[:beta_release],
-      external: true
+    ios_build_prechecks(skip_confirm: options[:skip_confirm], external: true) unless options[:skip_prechecks]
+    ios_build_preflight unless options[:skip_prechecks]
+
+    sentry_check_cli_installed
+    appstore_code_signing
+
+    gym(
+      scheme: 'WordPress',
+      workspace: WORKSPACE_PATH,
+      clean: true,
+      export_team_id: get_required_env('EXT_EXPORT_TEAM_ID'),
+      output_directory: BUILD_PRODUCTS_PATH,
+      derived_data_path: DERIVED_DATA_PATH,
+      export_options: { method: 'app-store' }
     )
 
-    ios_build_preflight
+    testflight(
+      skip_waiting_for_build_processing: true,
+      team_id: '299112',
```

Comment:
We have this in an environment variable, too.

```suggestion
      team_id: get_required_env('FASTLANE_ITC_TEAM_ID'),
```

I'm suggesting to use that only so that we DRY and centralize the value, even though, thinking about it, I don't think it'll ever change 🤔

Reply:
TIL we have that one (hidden in the `project.env` of mobile secret) 👍 

Yeah I agree that it's not much of a secret, and very unlikely it will ever change, so not worth having it as an ENV var compared to a regular constant. But getting rid of the too many ENV vars that shouldn't be ENV vars… is another PR 😅